l1-contracts/contracts/bridge/asset-router/AssetRouterBase.sol

```diff
@@ -0,0 +1,368 @@
+        bytes32 assetId = DataEncoding.encodeAssetId(block.chainid, _assetRegistrationData, sender);
+        if (!senderIsNTV && msg.sender != assetDeploymentTracker[assetId]) {
+            revert Unauthorized(msg.sender);
+        }
+        assetHandlerAddress[assetId] = _assetHandlerAddress;
+        assetDeploymentTracker[assetId] = msg.sender;
+        emit AssetHandlerRegisteredInitial(assetId, _assetHandlerAddress, _assetRegistrationData, sender);
+    }
+
+    function _setAssetHandlerAddress(bytes32 _assetId, address _assetAddress) internal {
+        assetHandlerAddress[_assetId] = _assetAddress;
+        emit AssetHandlerRegistered(_assetId, _assetAddress);
+    }
+
+    /// @dev Used to set the asset handler address on another chain. Not needed for NTV tokens.
+    /// @dev Currently only enabled on L1.
+    function _setAssetHandlerAddressOnCounterpart(
+        uint256 _chainId,
+        address _prevMsgSender,
+        bytes32 _assetId,
+        address _assetHandlerAddressOnCounterpart
+    ) internal virtual returns (L2TransactionRequestTwoBridgesInner memory request) {}
+
+    /// @inheritdoc IAssetRouterBase
+    function setAssetHandlerAddress(
+        uint256 _originChainId,
+        bytes32 _assetId,
+        address _assetAddress
+    ) external virtual override {}
+
+    /*//////////////////////////////////////////////////////////////
+                            INITIATTE DEPOSIT Functions
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IAssetRouterBase
+    function bridgehubDepositBaseToken(
+        uint256 _chainId,
+        bytes32 _assetId,
+        address _prevMsgSender,
+        uint256 _amount
+    ) public payable virtual override onlyBridgehub whenNotPaused {
+        _bridgehubDepositBaseToken(_chainId, _assetId, _prevMsgSender, _amount);
+    }
+
+    /// @notice Deposits the base token to the bridgehub internal. Used for access management.
+    function _bridgehubDepositBaseToken(
+        uint256 _chainId,
+        bytes32 _assetId,
+        address _prevMsgSender,
+        uint256 _amount
+    ) internal {
+        address assetHandler = assetHandlerAddress[_assetId];
+        if (assetHandler == address(0)) {
+            revert AssetHandlerDoesNotExist(_assetId);
+        }
+
+        // slither-disable-next-line unused-return
+        IAssetHandler(assetHandler).bridgeBurn{value: msg.value}({
+            _chainId: _chainId,
+            _msgValue: 0,
+            _assetId: _assetId,
+            _prevMsgSender: _prevMsgSender,
+            _data: abi.encode(_amount, address(0))
+        });
+
+        // Note that we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
+        emit BridgehubDepositBaseTokenInitiated(_chainId, _prevMsgSender, _assetId, _amount);
+    }
+
+    /// @inheritdoc IAssetRouterBase
+    function bridgehubDeposit(
```

Comment:
considering that this one is only really implemented for L1, does it make sense to move the code of the implementation ot the L1AssetRouter and keep the method fully without default impl here? Having a default implementation implies that the majority of contracts that inhreit this one use it, but this is not the case. 

We have only two imlpementors: L1AssetRouter and L2AssetRouter. If a function is implemented differently for both cases, it is much better to not have the standard implementation in base class to not confuse people